staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go

```diff
@@ -46,6 +49,15 @@ func WithPanicRecovery(handler http.Handler, resolver request.RequestInfoResolve
 				metrics.RecordRequestAbort(req, nil)
 			} else {
 				metrics.RecordRequestAbort(req, info)
+				if longRunning != nil && longRunning(req, info) {
+					// This was a long-running request such as a watch. Since
+					// these get ignored by WithTimeoutForNonLongRunningRequests,
```

Comment:
WithTimeoutForNonLongRunningRequests ignores watch because it enforces a deadline on non long-running requests only.

when a watch is ended gracefully (by the aggregated server) due to `timeoutSeconds` elapsing, I don't think it results in an `ErrAbortHandler` error:
https://github.com/kubernetes/kubernetes/blob/b15dfce6cbd0d5bbbcd6172cf7e2082f4d31055e/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go#L267-L268

Reply:
Ah, good to know that the "happy" timeout case shouldn't trigger this.

I mainly see it triggered by client-side timeouts.  Argo (I think) seems to time out all its watches after 10s, for example.

Reply:
Oh, is there an easy way for me to add the client identity info to the "Timeout or abort" log.  If that included the client's pod name, it'd help track down buggy clients a lot more easily(!)